src/main/java/uk/gov/hmcts/reform/sscs/ccd/service/UpdateCcdCaseService.java

```diff
@@ -151,4 +151,24 @@ public void setSupplementaryData(IdamTokens idamTokens, Long caseId, Map<String,
         ccdClient.setSupplementaryData(idamTokens, caseId, supplementaryData);
     }
 
+    /**
+     * Need to provide this so that recoverable/non-recoverable exception doesn't get wrapped in an IllegalArgumentException
+     */
+    @Recover
+    public int recover(RuntimeException exception, Long caseId, String eventType, String summary, String description, IdamTokens idamTokens, Consumer<SscsCaseDetails> mutator) {
+        log.error("In recover method(updateCaseV2 - Consumer) for caseId {} and eventType {} with exception {} ", caseId, eventType, exception.getMessage());
+        throw exception;
+    }
+
+    @Recover
+    public int recover(RuntimeException exception, Long caseId, String eventType, String summary, String description, IdamTokens idamTokens) {
+        log.error("In recover method(triggerCaseEventV2) for caseId {} and eventType {} with exception {} ", caseId, eventType, exception.getMessage());
+        throw exception;
+    }
+
+    @Recover
+    public int recover(RuntimeException exception, Long caseId, String eventType, IdamTokens idamTokens, Function<SscsCaseDetails, ?> mutator) {
```

Comment:
I think we can name these methods as recoverUpdateCaseV2Conditional and refer to that in `@retryable(recover="recoverUpdateCaseV2Conditional")`  ? Will look more meaningful and also if someone is copying, they won't forget to like we did :)

Reply:
Actually if we use annotation to direct it to method, then we might not need multiple methods with exact signature ? we can just use exception and case id  ( I think it might workout by parameter names)